backends-velox/src/main/scala/org/apache/gluten/extension/PullOutDuplicateProject.scala

```diff
@@ -60,13 +60,13 @@ object PullOutDuplicateProject extends Rule[SparkPlan] with PredicateHelper {
         bhj.streamedPlan.asInstanceOf[ProjectExecTransformer],
         bhj.references,
         pullOutAliases)
+      val newBhj = bhj.joinBuildSide match {
+        case BuildLeft => bhj.copy(right = newStreamedPlan)
+        case BuildRight => bhj.copy(left = newStreamedPlan)
+      }
       if (pullOutAliases.isEmpty) {
```

Comment:
Thanks for the fix, at present, pullOutAliases cannot determine whether rewriting is needed. We can separate the calculation of duplicates from rewriteProject and use duplicates to determine whether rewriting is needed.